src/main/java/swinglib/JButtonModal.java

```diff
@@ -0,0 +1,70 @@
+package swinglib;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import javax.swing.JButton;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+public class JButtonModal {
+
+  private final String title;
+  private final String toolTip;
+  private final Runnable actionListener;
+
+  public JButtonModal(final String title, final String toolTip, final Runnable actionListener) {
+    this.title = Preconditions.checkNotNull(title);
+    this.toolTip = toolTip;
+    this.actionListener = Preconditions.checkNotNull(actionListener);
+  }
+
+  public static JButtonModalBuilder builder() {
+    return new JButtonModalBuilder();
+  }
+
+  private JButton swingComponent() {
+    final JButton button = new JButton(title);
+    if (toolTip != null) {
+      button.setToolTipText(toolTip);
+    }
+    button.addActionListener(e -> actionListener.run());
+    return button;
+  }
+
+
+  public static class JButtonModalBuilder {
```

Comment:
Just thinking out loud...  Is there a need for both `JButtonModel` and `JButtonModelBuilder`?  It seems you only need the builder class, or at least `JButtonModel` could somehow be an implementation detail.

Also, do you perceive more factory methods in the future in addition to `swingComponent()`?  I didn't know if that's the reason you called it something other than `build()`.  The name `swingComponent()` wouldn't have been obvious to me.

If you can get rid of or hide the intermediate `JButtonModel` class, the API would be more like:

```java
final JButton button = JButtonBuilder.newInstance()
    .withTitle(...)
    .withTooltip(...)
    .withActionListener(...)
    .build();
```

I didn't test it, but the following compiles cleanly with 33% fewer lines (the `build()` method probably needs improvement):

```java
package swinglib;

import static com.google.common.base.Preconditions.checkNotNull;

import javax.swing.JButton;

import com.google.common.base.Preconditions;
import com.google.common.base.Strings;

public final class JButtonBuilder {
  private String title;
  private String toolTip;
  private Runnable actionListener;

  private JButtonBuilder() {}

  public static JButtonBuilder newInstance() {
    return new JButtonBuilder();
  }

  /** required - The text that will be on the button. */
  public JButtonBuilder withTitle(final String title) {
    Preconditions.checkArgument(!Strings.nullToEmpty(title).trim().isEmpty());
    this.title = title;
    return this;
  }

  /** optional, but potentially required in the future. This is the hover text when hovering on the button. */
  public JButtonBuilder withToolTip(final String toolTip) {
    Preconditions.checkArgument(!Strings.nullToEmpty(toolTip).trim().isEmpty());
    this.toolTip = toolTip;
    return this;
  }

  /** request, the event that occurs when the button is clicked */
  public JButtonBuilder withActionListener(final Runnable actionListener) {
    this.actionListener = checkNotNull(actionListener);
    return this;
  }

  public JButton build() {
    final JButton button = new JButton(Preconditions.checkNotNull(title));
    if (toolTip != null) {
      button.setToolTipText(toolTip);
    }
    button.addActionListener(e -> Preconditions.checkNotNull(actionListener).run());
    return button;
  }
}
```